botocore/credentials.py

```diff
@@ -791,6 +820,193 @@ def _assume_role_base_kwargs(self, config):
         return assume_role_kwargs
 
 
+def _role_selector(role_arn, roles):
+    """Given a roles list in the form of [{"RoleArn": "...", ...}, ...],
+    return the item which matches the role_arn, or None otherwise"""
+    chosen = [r for r in roles if r['RoleArn']==role_arn]
+    return chosen[0] if chosen else None
+
+
+class AssumeRoleWithSamlProvider(AssumeRoleProvider):
+    METHOD = 'assume-role-with-saml'
+    DISTINCTION_VAR = 'saml_endpoint'
+
+    def __init__(self, load_config, client_creator, cache, profile_name,
+                 role_selector=_role_selector,
+                 password_prompter=getpass.getpass,
+                 username_prompter=raw_input,
+                 authenticators=None):
+        """
+        :type authenticators: list
```

Comment:
Docstrings for the params would be nice here. Not completely obvious what all of the params represent.